[JIT] Support for ParameterDict getattr

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77143

Adds support for scripting ParameterDicts and getattr() on them. It does
not support iterating on ParameterDicts because torch/nn/container.py
implementation of ParameterDict.items() uses a generator, which is not
supported by torchscript. torch/nn/container.py would need to be updated
so that iter gets correctly registered in python_sugared_value.cpp

Added a test in test_module_containers.py

davidberard98 committed:
[JIT] Support for ParameterDict getattr

Adds support for scripting ParameterDicts and getattr() on them. It does
not support iterating on ParameterDicts because torch/nn/container.py
implementation of ParameterDict.items() uses a generator, which is not
supported by torchscript. torch/nn/container.py would need to be updated
so that iter gets correctly registered in python_sugared_value.cpp

Added a test in test_module_containers.py

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] Support for ParameterDict getattr"


Adds support for scripting ParameterDicts and getattr() on them. It does
not support iterating on ParameterDicts because torch/nn/container.py
implementation of ParameterDict.items() uses a generator, which is not
supported by torchscript. torch/nn/container.py would need to be updated
so that iter gets correctly registered in python_sugared_value.cpp

Added a test in test_module_containers.py

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] Support for ParameterDict getattr"


Adds support for scripting ParameterDicts and getattr() on them. It does
not support iterating on ParameterDicts because torch/nn/container.py
implementation of ParameterDict.items() uses a generator, which is not
supported by torchscript. torch/nn/container.py would need to be updated
so that iter gets correctly registered in python_sugared_value.cpp

Added a test in test_module_containers.py

[ghstack-poisoned]

eellison commented on torch/csrc/jit/python/python_sugared_value.cpp:
```diff
@@ -506,6 +515,33 @@ std::shared_ptr<SugaredDict> ModuleValue::getSugaredDict(
+    auto name_v =
+        std::make_shared<SimpleValue>(insertConstant(*m.graph(), name));
+    m.graph()->insertGetAttr(self_, name);
+    values.push_back(tryGetAttr(loc, m, name));
```
do we need to test if `tryGetAttr` is successful?

davidberard98 replied:
good idea, I added one

eellison reviewed (approved):
Nice, LGTM !!

davidberard98 committed:
Update on "[JIT] Support for ParameterDict getattr"


Adds support for scripting ParameterDicts and getattr() on them. It does
not support iterating on ParameterDicts because torch/nn/container.py
implementation of ParameterDict.items() uses a generator, which is not
supported by torchscript. torch/nn/container.py would need to be updated
so that iter gets correctly registered in python_sugared_value.cpp

Added a test in test_module_containers.py

[ghstack-poisoned]

davidberard98 commented:
@pytorchbot merge this
